skymap_scanner/utils/extract_json_message.py

```diff
@@ -229,36 +293,37 @@ def __extract_frame_packet(
         del baseline_GCD_framepacket
 
     LOGGER.info("Preprocessing event frames!")
-    if baseline_GCD is not None:
-        # frame_packet has GCD diff, provide baseline
-        frame_packet = prepare_frames(frame_packet, event_metadata, baseline_GCD_file, reco_algo, pulsesName)
-    else:
-        # frame_packet has either normal GCD or has been reassembled
-        frame_packet = prepare_frames(frame_packet, event_metadata, None, reco_algo, pulsesName)
+    # Uncompress GCD info and invoke `prepare_frames` traysegment provided by `reco_algo`
+    # - frame_packet has GCD diff => baseline_GCD_file is a path string
+    # - frame_packet has either normal GCD or has been reassembled => baseline_GCD_file is None
+    prepared_frame_packet = prepare_frames(frame_packet, event_metadata, baseline_GCD_file, reco_algo, pulses_name=pulses_name)
+
+    # Delete original frame packet.
+    del frame_packet
```

Comment:
I'm not a fan of using `del` to "switch main characters", ie `frame_packet` is no longer relevant now that `prepared_frame_packet` exists. It's a sign that the function is becoming bloated. The function could be made much more readable with additional helper functions considering its current size. A refactor may be beyond the scope of this PR, perhaps open a new issue?

Reply:
I agree! We can take a second look to see whether this can be divided into two functions.

Reply:
Added to #260